crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
I figured something like that; but the issue is that it sets the last ex_data value to NULL, which might have been set to something else. This causes a memory leak, and can potentially cause confusion in ex_data dup() functions that expect there to be ex_data present in the new object.